packages/shared/src/hooks/useTimedAnimation.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { isNullOrUndefined } from '../lib/func';
+
+interface UseTimedAnimation {
+  timer: number;
+  isAnimating: boolean;
+  endAnimation: () => void;
+}
+
+interface UseTimedAnimationProps {
+  autoEndAnimation: boolean;
+  animationDuration: number;
+  onAnimationEnd?: () => void;
+  onAnimationStop?: () => void;
+}
+
+const TEMPORARY_ID = 1;
+const INTERVAL_COUNT = 10;
```

Comment:
`INTERVAL_COUNT` name is not very explicit in this case.
I would suggest `PROGRESS_INTERVAL`. As it indicates how often we should trigger the progress update right?